core/server/models/member.js

```diff
@@ -0,0 +1,65 @@
+const ghostBookshelf = require('./base');
+
+ghostBookshelf.model('MemberPassword', ghostBookshelf.Model.extend({
+    tableName: 'members_passwords'
+}));
+
+const Member = ghostBookshelf.Model.extend({
+    tableName: 'members',
+
+    relationships: ['tokens', 'password'],
+
+    relationshipBelongsTo: {
+        tokens: 'tokens',
+        password: 'password'
+    },
+
+    emitChange: function emitChange(event, options) {
+        const eventToTrigger = 'member' + '.' + event;
+        ghostBookshelf.Model.prototype.emitChange.bind(this)(this, eventToTrigger, options);
+    },
+
+    onCreated: function onCreated(model, attrs, options) {
+        model.emitChange('added', options);
+    },
+
+    onUpdated: function onUpdated(model, attrs, options) {
+        model.emitChange('edited', options);
+    },
+
+    onDestroyed: function onDestroyed(model, options) {
+        model.emitChange('deleted', options);
+    },
+
+    toJSON: function toJSON(unfilteredOptions) {
+        var options = Member.filterOptions(unfilteredOptions, 'toJSON'),
+            attrs = ghostBookshelf.Model.prototype.toJSON.call(this, options);
+
+        // remove password hash and tokens for security reasons
+        delete attrs.password;
+        delete attrs.tokens;
+
+        return attrs;
+    },
+
+    password: function password() {
+        return this.hasOne('Credential').through('MemberPassword', 'id');
```

Comment:
I am not 💯 % if using `through` like this is correct, in particular about the second parameter `id`. This is based on the docs for it as well as the discussion on related issue [here](https://github.com/bookshelf/bookshelf/issues/1025#issuecomment-156868697), will need a bit of testing after connecting.

Reply:
The alternative as suggested on issue discussions is to use `belongsToMany`, which is relatively more straightforward, but returns a collection which in case of password is not needed. 😕